knip.ts

```diff
@@ -0,0 +1,136 @@
+      entry: ['tools/release/changelog-renderer.js'],
+      ignoreBinaries: [
+        // https://github.com/webpro/knip/issues/433
+        'stylelint',
+      ],
+    },
+    'packages/ast-spec': {
+      ignore: [
+        'src/**/fixtures/**',
+        'tests/*.type-test.ts',
+        // @typescript-eslint/typescript-estree is not listed in dependencies to avoid circular dependency errors
+        // You can check a more detailed explanation in this file
+        'tests/util/parsers/typescript-estree-import.ts',
+      ],
+    },
+    'packages/eslint-plugin': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/eslint-plugin-internal': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/integration-tests': {
+      ignore: ['fixtures/**'],
+    },
+    'packages/parser': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/repo-tools': {
+      ignoreDependencies: ['tsconfig.json'],
+    },
+    'packages/scope-manager': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/type-utils': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/typescript-estree': {
+      entry: ['src/use-at-your-own-risk.ts'],
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/website': {
+      entry: [
+        'docusaurus.config.mts',
+        'src/pages/**/*.tsx',
+
+        // imported in MDX docs
+        'src/components/RulesTable/index.tsx',
+        'src/components/TypeScriptOverlap/index.tsx',
+        'src/components/team/TeamBioList.tsx',
+        'src/components/team/TeamBioDivider.tsx',
+
+        // used by Docusaurus
+        'src/theme/CodeBlock/Content/String.tsx',
+        'src/theme/MDXComponents/index.tsx',
+        'src/theme/NotFound/Content/index.tsx',
+        'src/theme/prism-include-languages.js',
+      ],
+      ignoreDependencies: [
+        // used in MDX docs
+        'raw-loader',
+
+        // it's imported only as type (esquery types are forked and defined in packages/website/typings/esquery.d.ts)
+        'esquery',
+
+        '@babel/runtime',
+        '@docusaurus/mdx-loader',
+        '@docusaurus/types',
+        '@docusaurus/plugin-content-docs',
+        '@docusaurus/theme-search-algolia',
+        '@docusaurus/ExecutionEnvironment',
+        '@docusaurus/Link',
+        '@docusaurus/router',
+        '@docusaurus/useDocusaurusContext',
+        '@docusaurus/useBaseUrl',
+        '@docusaurus/BrowserOnly',
+        '@docusaurus/theme-classic',
+        '@generated/docusaurus.config',
+        '^@theme/.*',
+        '^@theme-original/.*',
+      ],
+    },
+    'packages/website-eslint': {
+      ignoreDependencies: [
+        // virtual module
+        'vt',
+      ],
+      entry: [
+        'src/index.js',
+        'src/mock/assert.js',
+        'src/mock/empty.js',
+        'src/mock/eslint-rules.js',
+        'src/mock/eslint.js',
+        'src/mock/lru-cache.js',
+        'src/mock/path.js',
+        'src/mock/typescript.js',
+        'src/mock/util.js',
```

Comment:
You know, as a followup I bet we could switch these over to plain `.ts` files, and remove the need for the `requireResolved` that's tripping up Knip... Not blocking at all IMO.